### ISSUE TITLE
PopupMenu: Inconsistent closing behavior

### ISSUE THREAD

**Issue by maltsevy** [infor-design/enterprise#7556]
**Describe the bug**
There is an inconsistent closing behavior for the popup menu.

**To Reproduce**

First example: When opening a popup menu for the first time and clicking on the item that has a submenu - the menu closes and the arrow stays
![image](https://github.com/infor-design/enterprise/assets/44119194/fa061534-2441-4b08-80c8-2a034c324337)
1. Refresh a screen
2. Open a popup menu
3. Click on an item that has a submenu
4. See that the whole popup menu disappears and the ui glitch happens (triangle stays)
5. Should just open the submenu.
The issue can be recreated on branch https://github.com/infor-design/enterprise-ng.git > /ids-enterprise-ng-demo/popupmenu
When the user clicks on the space inside the popupmenu - the menu closes, triangle stays
![image](https://github.com/infor-design/enterprise/assets/44119194/37593397-49f1-498a-bec0-336ccef6a72c)
![image](https://github.com/infor-design/enterprise/assets/44119194/b2a94e5b-0e91-4174-93af-0ea6875ed7ca)

Another issue with closing on hover:
1. Open a popup menu
2. Hover over an element with a submenu
3. Hover one element in the submenu and move the cursor outside of the component
4. See that the submenu stays open (should be closed)
5. Move the cursor back to the submenu and move out again 
7. See that submenu closes.
Can be recreated here: https://design.infor.com/code/ids-enterprise/latest/demo/components/popupmenu/example-submenus.html
(cursor is out of the submenu but the submenu stays open)
![image](https://github.com/infor-design/enterprise/assets/44119194/51f70ce2-9fc6-4605-81b5-e2da391dd2cb)

**Version**
- ids-enterprise: v4,14.7

**Platform**
 - Infor Application/Team Name: Infor WFM
 - OS Version: Windows 10
 - Browser Name: chrome
 - Browser Version: Version 113.0.5672.127 (Official Build) (64-bit)

**Comment by tmcconechy** [comment]
@maltsevy can you update the steps to be more clear? I dont really understand the bug (especially around "refreshing"). not seeing any issue on https://design.infor.com/code/ids-enterprise/latest/demo/components/popupmenu/example-submenus.html and the steps are too hard to follow?

**Comment by maltsevy** [comment]
@tmcconechy Updated the description, hope it will help

**Comment by maltsevy** [comment]
The first issue is the most important for our project, so I'll do my best to explain the problem clearly. Please let me know which part needs more clarification. Here's the problem: When you close the menu by clicking inside it, the menu itself disappears as expected. However, an "arrow" that shows the menu's position remains visible. This issue only occurs if the menu wasn't closed before. It seems that when you close the menu after clicking inside it, the position of the popup menu wrapper is not recalculated and updated correctly. As a result, the arrow indicating the menu's position doesn't hide properly when the menu is closed.
popupmenu-wrapper style attribute when the problem occurs:
![image](https://github.com/infor-design/enterprise/assets/44119194/32004c6d-c8c4-4a21-819e-87267851abd1)
![image](https://github.com/infor-design/enterprise/assets/44119194/615ff9b3-441b-4412-84e1-f729b1c8d6a0)
When works as expected:
![image](https://github.com/infor-design/enterprise/assets/44119194/4df00a15-e00f-411e-8fd9-3fd782ac32a8)
![image](https://github.com/infor-design/enterprise/assets/44119194/d54e5ccd-0f1e-40cc-be53-bcf76d4e538e)

**Comment by tmcconechy** [comment]
The problem is i cant reproduce that. Can you reproduce it on any example? https://main-enterprise.demo.design.infor.com/components/popupmenu

I see a link to a repo but it seems like a bad link. Here is a template for reporting angular issues you can fork and use: https://stackblitz.com/edit/ids-quick-start-1631?file=src%2Fapp%2Fapp.component.ts

Also did you confirm against the latest version?

**Comment by maltsevy** [comment]
I can't reproduce it on any of the examples on the https://main-enterprise.demo.design.infor.com/components/popupmenu
But I've forked the template on the stackblitz and managed to reproduce the problem
https://stackblitz.com/edit/ids-quick-start-1631-cfnfbk?file=src%2Fapp%2Fapp.component.html%3AL27
![ezgif-3-cbc6f8f749](https://github.com/infor-design/enterprise/assets/44119194/02a89bdf-2137-423e-8773-b02bb06def29)

**Comment by tmcconechy** [comment]
OK lets fix that but skip the other point.

**Comment by tmcconechy** [comment]
Note: Had to remove this as it caused serious issues https://github.com/infor-design/enterprise/pull/7765 not sure if this means this doesnt work. So try with 4.86 and let me know. I was unable to make the stackblitz work and confirm so might need a new stackblitz or simpler non-angular example if it does not
